ha/util/message_bus.py

```diff
@@ -0,0 +1,196 @@
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+import json
+from typing import Callable
+from threading import Thread
+from cortx.utils.message_bus import MessageBusAdmin
+from cortx.utils.message_bus.error import MessageBusError
+from cortx.utils.message_bus import MessageProducer
+from cortx.utils.message_bus import MessageConsumer
+
+class MessageBusProducer:
+    PRODUCER_METHOD = "sync"
+
+    def __init__(self, producer_id: str, message_type: str, partitions: int):
+        """
+        Register message types with message bus.
+
+        Args:
+            producer_id (str): producer id.
+            message_types (str): Message type.
+            partitions (int, optional): No. of partitions. Defaults to 1.
+
+        Raises:
+            MessageBusError: Message bus error.
+        """
+        self.producer = MessageProducer(producer_id=producer_id, message_type=message_type, method=MessageBusProducer.PRODUCER_METHOD)
+
+    def publish(self, message: any):
+        """
+        Produce message to message bus.
+
+        Args:
+            message (any): Message.
+            If msg is dict it will be dumped as json.
+            If msg is string then it will be send directly.
+            If message is list, it should have all string element, all items will be published.
+        """
+        if isinstance(message, dict):
+            self.producer.send([json.dumps(message)])
+        elif isinstance(message, str):
+            self.producer.send([message])
+        elif isinstance(message, list):
+            self.producer.send(message)
+        else:
+            raise Exception(f"Invalid type of message {message}")
+
+class CONSUMER_STATUS:
+    SUCCESS = "success"
+    FAILED = "failed"
+    FAILED_STOP = "failed_stop"
+    SUCCESS_STOP = "success_stop"
+
+class MessageBusConsumer:
+
+    def __init__(self, consumer_id: int, consumer_group: str, message_type: str,
+                callback: Callable, auto_ack: bool, offset: str):
+        """
+        Initalize consumer.
+
+        Args:
+            consumer_id (int): Consumer ID.
+            consumer_group (str): Consumer Group.
+            message_type (list): Message Type.
+            callback (Callable): function to get message.
+            auto_ack (bool, optional): Check auto ack. Defaults to False.
+            offset (str, optional): Offset for messages. Defaults to "earliest".
+        """
+        self.callback = callback
+        self.stop_thread = False
+        self.consumer_id = consumer_id
+        self.consumer_group = consumer_group
+        self.message_type = message_type
+        self.auto_ack = auto_ack
+        self.offset = offset
+
+    def run(self):
+        """
+        Overloaded of Thread.
+        1. return success and ack		SUCCESS
+        2. send same msg again			FAILED
+        3. ack and skip					SUCCESS
```

Comment:
I think SUCCESS is already handled at point 1. And if point 1 to be considered, what should be returned??

Reply:
return code is same but both case are different
1.  message received and processed correctly by caller
3. message received by caller but irrelavent still need to return success

will update cases properly in word

Reply:
updated

Reply:
Thanks. Got it.